swift/example_code/lambda/basics/lambda-basics/Sources/entry.swift

```diff
@@ -0,0 +1,540 @@
+
+            _ = try await lambdaClient.createFunction(
+                input: CreateFunctionInput(
+                    code: LambdaClientTypes.FunctionCode(zipFile: zipData),
+                    functionName: name,
+                    handler: "handle",
+                    role: roleArn,
+                    runtime: .providedal2
+                )
+            )
+        } catch {
+            return false
+        }
+        // snippet-end:[swift.lambda-basics.CreateFunction]
+
+        // Wait for a while to be sure the function is done being created.
+
+        let output = try await lambdaClient.waitUntilFunctionActiveV2(
+            options: WaiterOptions(
+                maxWaitTime: 20,
+                minDelay: 0.5,
+                maxDelay: 2
+            ),
+            input: GetFunctionInput(functionName: name)
+        )
+
+        switch output.result {
+            case .success:
+                return true
+            case .failure:
+                return false
+        }
+    }
+    // snippet-end:[swift.lambda-basics.CreateFunction.wait]
+
+    // snippet-start:[swift.lambda-basics.UpdateFunctionCode.wait]
+    /// Update the AWS Lambda function with new code to run when the function
+    /// is invoked.
+    /// 
+    /// - Parameters:
+    ///   - lambdaClient: The `IAMClient` to use.
+    ///   - name: The name of the AWS Lambda function to update.
+    ///   - path: The pathname of the Zip file containing the packaged Lambda
+    ///     function.
+    /// - Throws: `ExampleError.zipFileReadError`
+    /// - Returns: `true` if the function's code is updated successfully.
+    ///   Otherwise, returns `false`.
+    func updateFunctionCode(lambdaClient: LambdaClient, name: String,
+                            path: String) async throws -> Bool {
+        // snippet-start:[swift.lambda-basics.UpdateFunctionCode]
+        let zipUrl = URL(fileURLWithPath: path)
+        let zipData: Data
+
+        // Read the function's Zip file.
+
+        do {
+            zipData = try Data(contentsOf: zipUrl)
+        } catch {
+            throw ExampleError.zipFileReadError
+        }
+
+        // Update the function's code and wait for the updated version to be
+        // ready for use.
+
+        do {
+            _ = try await lambdaClient.updateFunctionCode(
+                input: UpdateFunctionCodeInput(
+                    functionName: name,
+                    zipFile: zipData
+                )
+            )
+        } catch {
+            return false
+        }
+        // snippet-end:[swift.lambda-basics.UpdateFunctionCode]
+
+        let output = try await lambdaClient.waitUntilFunctionUpdatedV2(
+            options: WaiterOptions(
+                maxWaitTime: 20,
+                minDelay: 0.5,
+                maxDelay: 2
+            ),
+            input: GetFunctionInput(
+                functionName: name
+            )
+        )
+
+        switch output.result {
```

Comment:
I see you use this pattern a ton, so no need to fix it everywhere all the time, but I'm a bit curious why WaiterOutput<T> doesn't have an `is_ok -> bool` utility?

Reply:
No, there isn't a feature like that. This switch construct is pretty common throughout the SDK.